### ISSUE TITLE
React DnD tutorial is hard to follow

### ISSUE THREAD

**Issue by TimmyGOGO** [react-dnd/react-dnd#1201]
**React DnD tutorial (chessboard example)**
I'm new to React (completed a few tutorials).
And it's still not clear for me what we are doing in 'Game' component. And how it links with index.js. Moreover, when I completed the tutorial, I saw that the knight couldn't move to anywhere (we could drag him, but after drop the app crashes: "Cannot have two HTML5 backends at the same time"). So I went to compare my code with example, but it has another code with different files in typescript! Okay, I rewrote my code according to the example and of course, change ReactDOM.render so as to it renders 'ChessboardTutorialApp'. 
And all I get is just a square 500x500 and nothing else.

**Probable solution**
It's necessary to add explanation to how interacts 'Game' with 'index' and other components (moreover this tutorial (http://react-dnd.github.io/react-dnd/docs/tutorial) partially written for beginners: 
> If you're new to React and know a thing or two about it, but yet have to gain some experience building components, this tutorial can also serve as an introduction to the React mode of thinking and the React workflow. 

And It would be nice to re-write the tutorial with typescript code in order to at least compare it with working version of the app.

**My project with chessboard from documentation-examples**
https://github.com/TimmyGOGO/react_dnd_tutorial

**Comment by TimmyGOGO** [comment]
**In order to run this example with Chessboard, you need to add to Board.tsx file next lines:**

`import HTML5Backend from 'react-dnd-html5-backend';
 import BoardSquare from './BoardSquare';
 ...
 export default DragDropContext(HTML5Backend)(Board);
`

**Comment by mikemeerschaert** [comment]
I'm having the same issue, and have the same opinion on the tutorial. I also ran into the "Error: Cannot have two HTML5 backends at the same time." There are no complete code examples in the tutorial, so I don't know if i missed something while following along, or if it was written for an older version of the library which has changed, or if i'm using the wrong version of react, etc. I tried the suggestion of adding the DragDropContext HOC to the board but it didn't fix the error, and the source code from github is all written in typescript and doesn't exactly mirror the code in the examples.

**Comment by rdurel423** [comment]
Move the provider to `index.js` and wrap your root component with it. It seems that as the tutorial is written every time the `Board` component re-renders it provides a new context with a new back end.

**Comment by dbolesta** [comment]
@rdurel423 This does not fix the issue. I am still receiving the same "Cannot have two HTML5 backends at the same time" error.

Full `index.js` file after moving provider:
```jsx
import React from 'react'
import ReactDOM from 'react-dom'
import Board from './Board'
import { observe } from './Game'
import { DragDropContextProvider } from 'react-dnd'
import HTML5Backend from 'react-dnd-html5-backend'

const root = document.getElementById('root')


observe(knightPosition =>
   ReactDOM.render(<DragDropContextProvider backend={HTML5Backend}><Board knightPosition={knightPosition} /></DragDropContextProvider>, root),
)
```

**Comment by jzombie** [comment]
+1 that it is hard to follow.

Nobody needs to have to read through a bunch of experimental code to see how a chessboard would be implemented when you could just show a working implementation of how the code works and let people modify it to suit their own needs.

I made it work for me after struggling with it for about 1.5 hours.

Furthermore, the toggle button for "hooks" and "decorators" seems broken....  clicking on either changes nothing at all.

While it seems to be a good library, the documentation is horrible.

**Comment by jzombie** [comment]
... and this "stale" bot needs an algorithm update.

Just because nobody addressed the bad documentation does not mean that the issue should be closed.
